api/v1alpha1/clustertemplate_types.go

```diff
@@ -50,23 +67,28 @@ type Templates struct {
 	PolicyTemplateDefaults string `json:"policyTemplateDefaults"`
 }
 
-// InputDataSchema encapsulates all the schemas required for ClusterTemplate.
-type InputDataSchema struct {
-	// ClusterInstanceSchema defines the subschema of ClusterInstance parameters that
+// TemplateParameterSchema encapsulates all the schemas required for ClusterTemplate.
+type TemplateParameterSchema struct {
```

Comment:
I think we need to change the TemplateParameterSchema type to runtime.RawExtension (full schema) in the ClusterTemplate, even though it’s for internal use. I realized that there are two new parameters(nodeClusterName and oCloudSiteId) and they should be part of the schema(input should come from clusterRequest). We will then need to ensure these four parameters are included in the schema with the expected types.

Reply:
For example,
```
spec:
  templateParameterSchema:
    properties:
      type: object
      required:
      - nodeClusterName
      - oCloudSiteId
      - policyTemplateParameters
      - clusterInstanceParameters
      nodeClusterName:
        type: string
      oCloudSiteId:
        type: string
      policyTemplateParameters:
        description: policyTemplateSchema defines the available parameters for cluster configuration
        properties:
          sriov-network-vlan-1:
            type: string
      clusterInstanceParameters:
        description: ClusterInstanceSpec defines the params that are allowed in the ClusterRequest spec.clusterInstanceInput
        properties:
          additionalNTPSources:
            description: AdditionalNTPSources is a list of NTP sources (hostname
              or IP) to be added to all cluster hosts. They are added to any NTP
              sources that were configured through other means.
            items:
              type: string
            type: array
        type: object
```

Reply:
Fixed